tremor-script/src/errors.rs

```diff
@@ -811,17 +830,77 @@ pub(crate) fn error_array_out_of_bound<'script, T, O: BaseExpr, I: BaseExpr>(
     outer: &O,
     inner: &I,
     path: &ast::Path<'script>,
-    r: IRange<usize>,
+    r: RangeExclusive<usize>,
+    len: usize,
     meta: &NodeMetas,
 ) -> Result<T> {
     let expr: Range = outer.extent(meta);
+    // FIXME: Why match on `path` when all arms do the same?!
+    // -> ideally: put the `path` into the `ErrorKind::ArrayOutOfRange`, handle in display
+    //        but: not trivial: `Path` is parametric in non-'static lifetime 'script
```

Comment:
Perhaps introducing a PathType enum would collapse this and allow generalizing it it some places?

```rust
enum PathType {
Local, 
Const,
Meta,
Event,
State,
}
impl Path {
  fn path_type(&self) -> PathType {...}
}
```

Reply:
Extracted it into a new issue: https://github.com/wayfair-tremor/tremor-runtime/issues/168